Casks/metaimage.rb

```diff
@@ -1,15 +1,21 @@
 cask "metaimage" do
-  version "1.9.8,6091"
+  version "1.9.8,b0f8d13a-8233-4fb9-acca-8246f4947a80"
   sha256 "12f5f575f679083706547c8adfd12c141c680a948313f3e47fd011842bf012d2"
 
-  url "https://neededapps.com/appcasts/metaimage/versions/MI-#{version.csv.first}.zip"
+  url "https://neededapps.nyc3.digitaloceanspaces.com/media/public/#{version.csv.second}.zip",
+      verified: "neededapps.nyc3.digitaloceanspaces.com/media/public/"
   name "MetaImage"
   desc "Editor to read, write and edit images metadata"
   homepage "https://neededapps.com/metaimage/"
 
   livecheck do
     url "https://neededapps.com/appcasts/metaimage/changelog.xml"
-    strategy :sparkle
+    strategy :sparkle do |item|
+      hex = item.url[%r{/([\h-]+)\.zip}i, 1]
+      next if hex.blank?
+
+      "#{item.short_version},#{hex}"
```

Comment:
```suggestion
    regex(%r{/([^/]+)\.zip$}i)
    strategy :sparkle do |item, regex|
      filename = item.url[regex, 1]
      next if filename.blank?

      "#{item.short_version},#{filename}"
```

Other than using `#regex` and passing it into the `strategy` block (the preferred approach), there's something to be said for loosening the regex in this particular instance. If upstream ever strays from a strictly hex/hyphen format, the `livecheck` block in this PR would simply break and give an `Unable to get versions` error. This would lead to missed versions until someone gets around to manually fixing the `livecheck` block.

The suggested regex is more likely to match a new filename/URL format (provided the extension doesn't change) and continue to provide version information. Even if the cask requires changes as part of being updated to a new version (e.g., new `url` scheme, updating the `livecheck` block, etc.), at least we would be aware of it.